cpp/dspace/doi.cc

```diff
@@ -0,0 +1,98 @@
+#include <iostream>
+#include <regex>
+#include <set>
+#include <unordered_map>
+#include <cstdlib>
+#include "DbConnection.h"
+#include "DbResultSet.h"
+#include "DbRow.h"
+#include "Downloader.h"
+#include "EmailSender.h"
+#include "HttpHeader.h"
+#include "JSON.h"
+#include "StlHelpers.h"
+#include "StringUtil.h"
+#include "VuFind.h"
+#include "util.h"
+
+using namespace std;
+
+const std::string DspaceServerURL("https://publikationen.uni-tuebingen.de/rest/items/");
+const std::string DOIKey("dc.relation.uri");
+const std::string PublicationTitleKey("dc.title");
+const std::vector<std::string> notification_email_addresses = { "andrii.lysohor@uni-tuebingen.de" };
```

Comment:
The notification email addresses still need to be defined, they will be most likely taken from a configuration file later